src/Consolonia.Core/Drawing/ConsoleBrush.cs

```diff
@@ -1,96 +1,14 @@
-            AvaloniaProperty.Register<ConsoleBrush, Color>(nameof(Color));
-
-        public static readonly StyledProperty<PixelBackgroundMode> ModeProperty =
-            AvaloniaProperty.Register<ConsoleBrush, PixelBackgroundMode>(nameof(Mode));
-
-        // ReSharper disable once UnusedMember.Global
-        public ConsoleBrush(Color color, PixelBackgroundMode mode) : this(color)
-        {
-            Mode = mode;
-        }
-
-        public ConsoleBrush(Color color)
-        {
-            Color = color;
-        }
-
-        public ConsoleBrush()
-        {
-        }
-
-        public PixelBackgroundMode Mode
-        {
-            get => GetValue(ModeProperty);
-            set => SetValue(ModeProperty, value);
-        }
-
-        public Color Color
-        {
-            get => GetValue(ColorProperty);
-            set => SetValue(ColorProperty, value);
-        }
-
-        public double Opacity => 1;
-        public ITransform Transform => null;
-        public RelativePoint TransformOrigin => RelativePoint.TopLeft;
-
-        /// <summary>
-        ///     Convert a IBrush to a Brush.
-        /// </summary>
-        /// <remarks>
-        ///     NOTE: If it's a ConsoleBrush it will be passed through unchanged, unless mode is set then it will convert
-        ///     consolebrush to mode
-        /// </remarks>
-        /// <param name="brush"></param>
-        /// <param name="mode">Default is Colored.</param>
-        /// <returns></returns>
-        public static ConsoleBrush FromBrush(IBrush brush, PixelBackgroundMode? mode = null)
-        {
-            ArgumentNullException.ThrowIfNull(brush, nameof(brush));
-
-            switch (brush)
-            {
-                case ConsoleBrush consoleBrush:
-                    if (mode != null && consoleBrush.Mode != mode)
-                        return new ConsoleBrush(consoleBrush.Color, mode.Value);
-                    return consoleBrush;
-
-                case LineBrush lineBrush:
-                    switch (lineBrush.Brush)
-                    {
-                        case ConsoleBrush consoleBrush:
-                            return consoleBrush;
-                        case ISolidColorBrush br:
-                            return new ConsoleBrush(br.Color, mode ?? PixelBackgroundMode.Colored);
-                        default:
-                            ConsoloniaPlatform.RaiseNotSupported(6);
-                            return null;
-                    }
-
-                case ISolidColorBrush solidBrush:
-                    return new ConsoleBrush(solidBrush.Color, mode ?? PixelBackgroundMode.Colored);
-
-                default:
-                    ConsoloniaPlatform.RaiseNotSupported(6);
-                    return null;
-            }
-        }
-
-        public static ConsoleBrush FromPosition(IBrush brush, int x, int y, int width, int height)
+        public static Color FromPosition(IBrush brush, int x, int y, int width, int height)
```

Comment:
do you think using 'this' would be bad?  Or do you prefer them as static methods? 
brush.FromPosition(x,y,width, height)
vs
BrushExtensions.FromPosition(brush, x,y,heigh)?

Reply:
fixed